wsmaster/che-core-sql-schema/src/main/resources/che-schema/5.6.0/postgresql/3__create_factory_name_plus_userid_index.sql

```diff
@@ -0,0 +1,46 @@
+--
+--  [2012] - [2017] Codenvy, S.A.
+--  All Rights Reserved.
+--
+-- NOTICE:  All information contained herein is, and remains
+-- the property of Codenvy S.A. and its suppliers,
+-- if any.  The intellectual and technical concepts contained
+-- herein are proprietary to Codenvy S.A.
+-- and its suppliers and may be covered by U.S. and Foreign Patents,
+-- patents in process, and are protected by trade secret or copyright law.
+-- Dissemination of this information or reproduction of this material
+-- is strictly forbidden unless prior written permission is obtained
+-- from Codenvy S.A..
+--
+
+
+-- Generate names where they are not set.
+UPDATE che_factory
+SET name = concat('f', right(id, 9)) WHERE name IS NULL OR name = '';
+
+-- Make names unique for the same user, e.g if there is more than one factory with same name and user,
+-- leave first one and rename others.
+WITH dupes AS
```

Comment:
Yes I can move naming uniques into 1.1 in codenvy, then 1.2 will do the migration, and costraint will be just in `che_factory` table definition.